bin/main.ml

```diff
@@ -341,30 +345,85 @@ let loop_gameplay (evt : LTerm_event.t) (current_state : gameplay_state)
       | _ -> LoopResultContinue)
   | _ -> LoopResultContinue
 
-let loop_victory (evt : LTerm_event.t) (_ : victory_state) : loop_result
-    =
+let loop_victory (evt : LTerm_event.t) (victory_state : victory_state) :
+    loop_result =
   match evt with
   | LTerm_event.Key _ ->
       let dictionary = Lazy.force global_dictionary in
+      LoopResultUpdateState
+        (Gameplay
+           (blank_gameplay_state victory_state.players dictionary))
+  | _ -> LoopResultContinue
 
-      (* Define players *)
-      let player1 =
-        { name = "P1"; points = 0; letters = new_deck () }
-      in
-      let player2 =
-        { name = "P2"; points = 0; letters = new_deck () }
-      in
-      let player3 =
-        { name = "P3"; points = 0; letters = new_deck () }
+let loop_title (evt : LTerm_event.t) (title_state : title_state) :
+    loop_result =
+  match evt with
+  | LTerm_event.Key { code = LTerm_key.Char c; control = false; _ } ->
+      let c = UChar.char_of c in
+      let c = String.make 1 c in
+      let rec update_last_player_name players =
+        match players with
+        | [] -> [ { name = c; points = 0; letters = new_deck () } ]
+        | [ last ] -> [ { last with name = last.name ^ c } ]
+        | current :: remainder ->
+            current :: update_last_player_name remainder
       in
-      let player4 =
-        { name = "P4"; points = 0; letters = new_deck () }
+      LoopResultUpdateState
+        (Title { players = update_last_player_name title_state.players })
+  | LTerm_event.Key { code = LTerm_key.Char c; control = true; _ }
+    when UChar.char_of c == 'c' ->
+      LoopResultExit
+  | LTerm_event.Key { code = LTerm_key.Enter; _ } ->
+      if
+        List.length title_state.players > 1
+        && List.for_all
+             (fun player -> player.name != "")
+             title_state.players
+      then
+        LoopResultUpdateState
+          (Gameplay
+             (blank_gameplay_state title_state.players
+                (Lazy.force global_dictionary)))
+      else LoopResultContinue
+  | LTerm_event.Key { code = LTerm_key.Tab; _ } ->
+      LoopResultUpdateState
+        (Title
+           {
+             players =
+               title_state.players
+               @ [ { name = ""; points = 0; letters = new_deck () } ];
+           })
+  | LTerm_event.Key { code = LTerm_key.Delete; _ } ->
+      let rec remove_last_player players =
+        match players with
+        | [] -> []
+        (* if only one player remains, clear their name instead of
+           deleting them *)
+        | [ player ] -> [ { player with name = "" } ]
+        | current :: remainder ->
+            current :: remove_last_player remainder
```

Comment:
I believe this clears the last player's name no matter what. So players aren't getting deleted rather the last player is just getting their name cleared.

Reply:
fixed